Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,675 @@
+                return this._boundingSphere;
+            }
+        },
+        allowPicking : {
+            get : function() {
+                return this._allowPicking;
+            }
+        },
+        instancesLength : {
+            get : function() {
+                return this._instances.length;
+            }
+        },
+        ready : {
+            get : function() {
+                return this._ready;
+            }
+        },
+        readyPromise : {
+            get : function() {
+                return this._readyPromise;
+            }
+        }
+    });
+
+    function getInstancedUniforms(collection, programName) {
+        if (defined(collection._instancedUniformsByProgram)) {
+            return collection._instancedUniformsByProgram[programName];
+        }
+
+        var instancedUniformsByProgram = {};
+
+        var modelSemantics = ['MODEL', 'MODELVIEW', 'MODELVIEWPROJECTION', 'MODELINVERSE', 'MODELVIEWINVERSE', 'MODELVIEWPROJECTIONINVERSE', 'MODELINVERSETRANSPOSE', 'MODELVIEWINVERSETRANSPOSE'];
+        var supportedSemantics = ['MODELVIEW', 'MODELVIEWPROJECTION', 'MODELVIEWINVERSETRANSPOSE'];
+
+        var gltf = collection._model.gltf;
+        var techniques = gltf.techniques;
+        for (var techniqueName in techniques) {
+            if (techniques.hasOwnProperty(techniqueName)) {
+                var technique = techniques[techniqueName];
+                var parameters = technique.parameters;
+                var pass = technique.passes[technique.pass];
+                var instanceProgram = pass.instanceProgram;
+                var instanceProgramName = instanceProgram.program;
+                // Different techniques may share the same program, skip if already processed.
+                // This assumes techniques that share a program do not declare different semantics for the same uniforms.
+                if (!defined(instancedUniformsByProgram[instanceProgramName])) {
+                    var uniformMap = {};
+                    instancedUniformsByProgram[instanceProgramName] = uniformMap;
+                    var uniforms = instanceProgram.uniforms;
+                    for (var uniformName in uniforms) {
+                        if (uniforms.hasOwnProperty(uniformName)) {
+                            var parameterName = uniforms[uniformName];
+                            var parameter = parameters[parameterName];
+                            var semantic = parameter.semantic;
+                            if (defined(semantic) && (modelSemantics.indexOf(semantic) > -1)) {
+                                if (supportedSemantics.indexOf(semantic) > -1) {
+                                    uniformMap[uniformName] = semantic;
+                                } else {
+                                    // TODO : Maybe fail silently and fallback to a different approach
+                                    throw new RuntimeError('Model shader cannot be optimized for instancing');
```

Comment:
Throw a `DeveloperError` and document it and the supported instance-able semantics.  Make the error message precise: mention the name of the program, parameter, and semantic.